core/src/test/java/org/apache/calcite/tools/FrameworksTest.java

```diff
@@ -439,6 +441,36 @@ private void checkTypeSystem(final int expected, FrameworkConfig config) {
         CalciteSystemProperty.DEBUG.value());
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-7039">[CALCITE-7039]
+   * PlannerImpl supports custom Volcano RuleSet</a>. */
+  @Test void testAddCustomRules() throws Exception {
+    Table table = new TableImpl();
+    final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+    SchemaPlus schema = rootSchema.add("x", new AbstractSchema());
+    schema.add("MYTABLE", table);
+    List<RelTraitDef> traitDefs = new ArrayList<>();
+    traitDefs.add(ConventionTraitDef.INSTANCE);
+    traitDefs.add(RelDistributionTraitDef.INSTANCE);
+    SqlParser.Config parserConfig =
+            SqlParser.Config.DEFAULT
+                    .withCaseSensitive(false);
+
+    final FrameworkConfig config = Frameworks.newConfigBuilder()
```

Comment:
This test case doesn't specify whether the newly added plan has taken effect or not.

Reply:
Currently, all EnumerableRules are added by default, so the plan can be executed normally. I think leaving RuleSet empty can reflect the impact.

Reply:
How about I add one more case, that is, add a minimum set of executable EnumerableRules?

Reply:
```
final FrameworkConfig config = Frameworks.newConfigBuilder()
            .parserConfig(parserConfig)
            .defaultSchema(schema)
            .traitDefs(traitDefs)
            .build();
```
According to your description, this should execute the sql no exception.

Reply:
It's quite strange that we need to compare with testUpdate to check whether this test case is complete.

Reply:
> How about I add one more case, that is, add a minimum set of executable EnumerableRules?

@NobiGo I think I understand what you mean. I think this should be used as a positive test and prove that.